sdk/ml/azure-ai-ml/azure/ai/ml/entities/_registry/registry_support_classes.py

```diff
@@ -210,10 +178,89 @@ def _to_rest_object(self) -> RestRegistryRegionArmDetails:
                 acr) for acr in self.acr_config]
         storages = []
         if self.storage_config:
-            storages = [SystemCreatedStorageAccount._to_rest_object(
-                storage) for storage in self.storage_config]
+            storages = self._storage_config_to_rest_object()
         return RestRegistryRegionArmDetails(
             acr_details=converted_acr_details,
             location=self.location,
             storage_account_details=storages,
         )
+
+
+    def _storage_config_to_rest_object(self) -> List[RestStorageAccountDetails]:
```

Comment:
Sorry may be I am missing something here, on why these two methods cannot be part of SystemCreatedStorageAccount

Reply:
Two reasons:
 - These functions need the adaptability to deal with user-created storage accounts as well. 
 - Due to the replication count feature, the parity between SystemCreatedStorageAccount and the rest equivalent is not 1-1, so I moved this logic one layer up to deal with that more easily.